inspirehep/modules/workflows/views.py

```diff
@@ -376,3 +401,180 @@ def robotupload_callback():
+
+        Examples:
+            An example of successful call:
+
+                $ curl \\
+                    http://web:5000/callback/workflows/resolve_validation_errors \\
+                    -H "Host: localhost:5000" \\
+                    -H "Content-Type: application/json" \\
+                    -d '{
+                        "_extra_data": {
+                            ... extra data content
+                        },
+                        "id": 910648,
+                        "metadata": {
+                            "$schema": "https://labs.inspirehep.net/schemas/records/hep.json",
+                            ... record content
+                        }
+                    }'
+
+            The response:
+
+                HTTP 200 OK
+
+                {"mesage": "Workflow 910648 validated, continuing it."}
+
+
+            A failed example:
+
+                $ curl \\
+                    http://web:5000/callback/workflows/resolve_validation_errors \\
+                    -H "Host: localhost:5000" \\
+                    -H "Content-Type: application/json" \\
+                    -d '{
+                        "_extra_data": {
+                            ... extra data content
+                        },
+                        "id": 910648,
+                        "metadata": {
+                            "$schema": "https://labs.inspirehep.net/schemas/records/hep.json",
+                            ... record content
+                        }
+                    }'
+
+            The error response will contain the workflow that was passed, with the
+            new validation errors:
+
+                HTTP 400 Bad request
+
+                {
+                    "_extra_data": {
+                        "validatior_errors": [
+                            {
+                                "path": ["path", "to", "error"],
+                                "message": "required: ['missing_key1', 'missing_key2']"
+                            }
+                        ],
+                        ... rest of extra data content
+                    },
+                    "id": 910648,
+                    "metadata": {
+                        "$schema": "https://labs.inspirehep.net/schemas/records/hep.json",
+                        ... record content
+                    }
+                }
+        """
+        workflow_data = _validate_workflow_payload(
+            ['metadata', '_extra_data', 'id'])
+        workflow_id = workflow_data['id']
+
+        try:
+            workflow = workflow_object_class.get(workflow_id)
+        except WorkflowsMissingObject:
+            raise CallbackWorkflowNotFoundError(workflow_id)
+
+        workflow.extra_data.pop('callback_url', None)
+        workflow.extra_data.pop('validation_errors', None)
+
+        workflow.data = workflow_data['metadata']
+        workflow.save()
+        db.session.commit()
+
+        if workflow.status != ObjectStatus.COMPLETED:
```

Comment:
You might want to do this check before changing the workflow (a few lines before this one).

Reply:
hmm, also even rise a different error message, like 'workflow is in completed state' or similar.

Reply:
Also, it should make sure that the workflow is halted because of validation errors :S, not sure if that's easy, but otherwise, if you might un halt a workflow that is waiting for something else. Maybe we could check the workflow's 'callback_url' to make sure that it's us, or (better but longer) finally define the halted statuses properly and check for that.

Reply:
I'm a bit confused, the status when we call the validation error is ``ERROR`` right?